presto-main/src/main/java/com/facebook/presto/memory/ClusterMemoryLeakDetector.java

```diff
@@ -45,79 +41,54 @@
     // we check to see whether some time has passed after the query finishes to claim that it is leaked.
     private static final int DEFAULT_LEAK_CLAIM_DELTA_SEC = 60;
 
+    private static final long BYTES_TO_GIGABYTES = (long) 1e9;
+
     @GuardedBy("this")
     private Set<QueryId> leakedQueries;
 
-    /**
-     * @param queryInfoSupplier All queries that the coordinator knows about.
-     * @param queryMemoryReservations The memory reservations of queries in the GENERAL cluster memory pool.
-     */
-    void checkForMemoryLeaks(Supplier<List<BasicQueryInfo>> queryInfoSupplier, Map<QueryId, Long> queryMemoryReservations)
-    {
-        requireNonNull(queryInfoSupplier);
-        requireNonNull(queryMemoryReservations);
-
-        Map<QueryId, BasicQueryInfo> queryIdToInfo = Maps.uniqueIndex(queryInfoSupplier.get(), BasicQueryInfo::getQueryId);
-
-        Map<QueryId, Long> leakedQueryReservations = queryMemoryReservations.entrySet()
-                .stream()
-                .filter(entry -> entry.getValue() > 0)
-                .filter(entry -> isLeaked(queryIdToInfo, entry.getKey()))
-                .collect(toImmutableMap(Entry::getKey, Entry::getValue));
-
-        if (!leakedQueryReservations.isEmpty()) {
-            log.debug("Memory leak detected. The following queries are already finished, " +
-                    "but they have memory reservations on some worker node(s): %s", leakedQueryReservations);
-        }
-
-        synchronized (this) {
-            leakedQueries = ImmutableSet.copyOf(leakedQueryReservations.keySet());
-        }
-    }
+    @GuardedBy("this")
+    private long leakedGb;
```

Comment:
why not leak it as bytes and do the convertion in monitoring tier? We do that all other places in the code base.